### PR TITLE
feat: add mount option filtering to disk plugin

### LINKED ISSUE #10986: inputs.disk log spamming with encrypted disks decrypted using zuluCrypt
### Relevant telegraf.conf

```toml
# Read metrics about disk usage by mount point
[[inputs.disk]]
  ## By default stats will be gathered for all mount points.
  ## Set mount_points will restrict the stats to only the specified mount points.
  # mount_points = ["/"]

  ## Ignore mount points by filesystem type.
  ignore_fs = ["tmpfs", "devtmpfs", "devfs", "iso9660", "overlay", "aufs", "squashfs", "cgroup", "sysfs", "proc", "securityfs", "devpts", "pstore", "configfs", "hugetlbfs", "debugfs", "selinuxfs", "mqueue", "rpc_pipefs", "nsfs", "tracefs", "fuse.portal" ]
```

### Logs from Telegraf

```text
Apr 14 15:34:50 host.example.com telegraf[701511]: 2022-04-14T13:34:50Z I! Starting Telegraf 1.22.1
Apr 14 15:34:50 host.example.com telegraf[701511]: 2022-04-14T13:34:50Z I! Loaded inputs: chrony conntrack cpu disk diskio hddtemp kernel mem netstat processes sensors swap system
Apr 14 15:34:50 host.example.com telegraf[701511]: 2022-04-14T13:34:50Z I! Loaded aggregators:
Apr 14 15:34:50 host.example.com telegraf[701511]: 2022-04-14T13:34:50Z I! Loaded processors:
Apr 14 15:34:50 host.example.com telegraf[701511]: 2022-04-14T13:34:50Z I! Loaded outputs: influxdb
Apr 14 15:34:50 host.example.com telegraf[701511]: 2022-04-14T13:34:50Z I! Tags enabled: host=host.example.com
Apr 14 15:34:50 host.example.com systemd[1]: Started The plugin-driven server agent for reporting metrics into InfluxDB.
Apr 14 15:34:50 host.example.com telegraf[701511]: 2022-04-14T13:34:50Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"host.example.com", Flush Interval:10s
Apr 14 15:35:00 host.example.com telegraf[701511]: 2022-04-14T13:35:00Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:35:10 host.example.com telegraf[701511]: 2022-04-14T13:35:10Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:35:20 host.example.com telegraf[701511]: 2022-04-14T13:35:20Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:35:30 host.example.com telegraf[701511]: 2022-04-14T13:35:30Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:35:40 host.example.com telegraf[701511]: 2022-04-14T13:35:40Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:35:50 host.example.com telegraf[701511]: 2022-04-14T13:35:50Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:36:00 host.example.com telegraf[701511]: 2022-04-14T13:36:00Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:36:10 host.example.com telegraf[701511]: 2022-04-14T13:36:10Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:36:20 host.example.com telegraf[701511]: 2022-04-14T13:36:20Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:36:30 host.example.com telegraf[701511]: 2022-04-14T13:36:30Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
Apr 14 15:36:40 host.example.com telegraf[701511]: 2022-04-14T13:36:40Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/media/private/root/sdb1"): permission denied
```

### System info

Telegraf 1.22.1, Fedora 35

### Docker

_No response_

### Steps to reproduce

I use [zuluCrypt](https://mhogomchungu.github.io/zuluCrypt) to decrypt a disk with the command

```bash
zuluCrypt-cli -o -d /dev/sdb1 -M -m sdb1
```

and this (specifically the `-M` option) causes the decrypted file system to have *two* mount points, where telegraf does not have permission to the private mount:

```bash
$ sudo -u telegraf cat /proc/self/mounts | grep sdb1
/dev/mapper/zuluCrypt-0-NAAN-sdb1-782698459 /run/media/private/root/sdb1 ext4 rw,seclabel,nosuid,nodev,relatime 0 0
/dev/mapper/zuluCrypt-0-NAAN-sdb1-782698459 /run/media/public/sdb1 ext4 rw,seclabel,nosuid,nodev,relatime 0 0
$ sudo -u telegraf stat /run/media/private/root/sdb1
stat: cannot statx '/run/media/private/root/sdb1': Permission denied
$ sudo -u telegraf stat /run/media/public/sdb1
  File: /run/media/public/sdb1
  Size: 4096            Blocks: 8          IO Block: 4096   directory
Device: fd01h/64769d    Inode: 2           Links: 5
Access: (0777/drwxrwxrwx)  Uid: ( ...)   Gid: ( ...)
Context: system_u:object_r:container_file_t:s0
Access: 2022-04-14 15:35:35.195353161 +0200
Modify: 2020-10-13 02:47:49.450933429 +0200
Change: 2022-04-11 21:06:11.437295638 +0200
 Birth: 2017-08-17 21:14:39.000000000 +0200
$
```

### Expected behavior

While zuluCrypt perhaps is too niche to expect to support out of the box, I most certainly expect to be able to manually mitigate the issue when discovered. Maybe with a `ignore_mount_points` option?

I can live with a single, static `/run/media/private/root/sdb1` ignore entry since I currently only have one disk I decrypt with zuluCrypt, although being able to filter on `/run/media/private/*` would be the best.

### Actual behavior

Telegraf is currently unusable since it spams the log constantly.

### Additional info

This is very similar to https://github.com/influxdata/telegraf/issues/10897, however this cannot be filtered by file system type.

### PR DESCRIPTION
**Pull request by nferch**
### Required for all PRs:

- [X] Updated associated README.md.
- [X] Wrote appropriate unit tests.
- [X] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #10986 and #10897

Nomad (and possibly other software) will create bind mounts in locations that Telegraf running as a non-root user cannot inspect. This creates an `ignore_mount_opts` option to the disk plugin that can ignore these mounts, which are unlikely to generate useful metrics anyway.

### CODE REVIEW

**Comment by nferch** [comment]
!signed-cla

File: plugins/inputs/disk/README.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ Note that `used_percent` is calculated by doing `used / (used + free)`, _not_
 
   ## Ignore mount points by filesystem type.
   ignore_fs = ["tmpfs", "devtmpfs", "devfs", "iso9660", "overlay", "aufs", "squashfs"]
+
+  ## Ignore mount points by mount options.
+  ignore_mount_opts = ["bind"]
```
Comment (powersj) [inline plugins/inputs/disk/README.md:22]: The default config options should match what is used and be commented out. I do not believe we should be removing any mount with the `bind` option by default and instead leave that to the user.

I have never seen a `bind` option on a mount and it appears that this is something that `gopsutil` [adds to the options](https://github.com/shirou/gopsutil/blob/cf6309316c89a96a0a4b848185b32830a6e8175e/disk/disk_linux.go#L323). It makes me a little wary of making this change if we ever go away from gopsutil, even if that is unlikely.

I think this is worth a comment in the docs as this is pretty non-standard:

```suggestion
  ## Ignore mount points by mount options.
  ## The `mount` command reports options of all mounts in parathesis.
  ## Bind mounts can be ignored with the special `bind` option.
  # ignore_mount_opts = []
```

**Review by powersj — CHANGES_REQUESTED** [review]
Hi,

Thanks for looking at this! I've added a few comments below.

Thanks

File: plugins/inputs/disk/README.md (hunk 1 above)
Comment (nferch) [reply plugins/inputs/disk/README.md:22]: Makes sense. 

Good callout on being conservative with bind mounts. FWIW from the mount(8) manpage it seems that it's possible, but not common, to specify `bind` as a mount option, although mount(8) handles it on its own and doesn't pass it to the kernel directly. So I think it's a bit of a convention to include `bind` along with other mount options.

**Review by powersj — APPROVED** [review]
Thanks again!

**Comment by reimda** [comment]
Is the purpose of this change to stop seeing "error getting disk usage" messages in logs? https://github.com/influxdata/telegraf/pull/10925 was merged recently and took a different approach to fixing that problem. It changed the error message to a debug message.

Is the ignore_mount_opts option still going to be useful now that the message is downgraded to debug?

**Comment by nferch** [comment]
> Is the purpose of this change to stop seeing "error getting disk usage" messages in logs? #10925 was merged recently and took a different approach to fixing that problem. It changed the error message to a debug message.
> 
> Is the ignore_mount_opts option still going to be useful now that the message is downgraded to debug?

The debug messages were a symptom of the problem. I would rather Telegraph not attempt to collect disk usage for bind mounts at all, which could be quite numerous depending on the workload.

**Review by reimda — APPROVED** [review]

**Comment by reimda** [comment]
> The debug messages were a symptom of the problem. I would rather Telegraph not attempt to collect disk usage for bind mounts at all, which could be quite numerous depending on the workload.

I agree this is a cleaner fix, and it's better not to hide the symptoms. Thanks!

**Pull request merged**